src/python/pose_format/bin/directory.py

```diff
@@ -4,11 +4,43 @@
 from typing import List
 import logging
 from tqdm import tqdm
+import multiprocessing
+from multiprocessing import Pool
+import multiprocessing.pool as mpp
+import psutil
+import os
+
+# os.system("taskset -p 0xff %d" % os.getpid())
 
 # Note: untested other than .mp4. Support for .webm may have issues: https://github.com/sign-language-processing/pose/pull/126
 SUPPORTED_VIDEO_FORMATS = [".mp4", ".mov", ".avi", ".mkv", ".flv", ".wmv", ".webm"]
 
 
+# https://stackoverflow.com/questions/57354700/starmap-combined-with-tqdm
+
+def istarmap(self, func, iterable, chunksize=1):
+    """starmap-version of imap
+    """
+    self._check_running()
+    if chunksize < 1:
+        raise ValueError(
+            "Chunksize must be 1+, not {0:n}".format(
+                chunksize))
+
+    task_batches = mpp.Pool._get_tasks(func, iterable, chunksize)
+    result = mpp.IMapIterator(self)
+    self._taskqueue.put(
+        (
+            self._guarded_task_generation(result._job,
+                                          mpp.starmapstar,
+                                          task_batches),
+            result._set_length
+        ))
+    return (item for chunk in result for item in chunk)
+
+mpp.Pool.istarmap = istarmap
```

Comment:
What does this do? Are we redefining a function from a library?

Reply:
Ah, it's from https://stackoverflow.com/questions/57354700/starmap-combined-with-tqdm